### PR TITLE
Feature/flowmap

### PR DESCRIPTION
**Pull request by SiggyF**
Implement flowmap conversion.

### CODE REVIEW

File: dgds_backend/config_data/datasets_access.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     },
     "rasterService": {
       "url": "https://dgds-test-dot-hydro-engine.appspot.com/get_glossis_data",
-      "featureinfo_url": "https://dgds-test-dot-hydro-engine.appspot.com/get_feature_info",
+      "featureInfoUrl": "https://dgds-test-dot-hydro-engine.appspot.com/get_feature_info",
```
Comment (evetion) [inline dgds_backend/config_data/datasets_access.json]: This would require frontend changes, which branch/issue do we have for this?

File: dgds_backend/config_data/example.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            "type": "vector"
+        },
+        {
+            "id": "GLOSSIS-earthengine",
+            "type": "raster",
+            "url": "http://pl-tc012.xtr.deltares.nl:8080/FewsWebServices/wms?service=WMS&request=GetMap&version=1.3&layers=Current%202DH&styles=&format=image%2Fpng&transparent=true&crs=EPSG%3A3857&time=##TIME##&uppercase=false&width=256&height=256&bbox={bbox-epsg-3857}"
+        }
+    ],
+    "metadata": {
+    }
```
Comment (evetion) [inline dgds_backend/config_data/example.json]: Missing comma.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# http://editorconfig.org
```
Comment (evetion) [inline .editorconfig]: Always good to unify our styling :)

File: datasets/glossis/ee-output-tiles.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# This script syncs buckets
```
Comment (evetion) [inline datasets/glossis/ee-output-tiles.sh]: It doesn't only sync data. Could you extend the description?

File: notebooks/glossis-currents.ipynb [hunk 5]
```diff
@@ -0,0 +1,472 @@
+{
```
Comment (evetion) [inline notebooks/glossis-currents.ipynb]: This is just here for reference and not used right?

File: datasets/glossis/ee-output-tiles.sh [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+gsutil -m rsync gs://dgds-flowmap/ee-output .
+# Make a tiles  directory
+mkdir tiles
+# cd  to tiles
+pushd tiles
+# loop over all  output files
+for f in ../glossis-current-*.tif*
+do
+    # convert to tiles
+    python ~/src/dgds-backend/datasets/flowmap/glossis2wgs84-tiles.py --max-zoom $maxzoom $f
```
Comment (evetion) [inline datasets/glossis/ee-output-tiles.sh]: This needs to be updated.

File: dgds_backend/config_data/datasets.json [hunk 7]
```diff
@@ -25,6 +25,14 @@
       "units": "m/s",
       "timeSpan": "Live",
       "toolTip": "Water level, storm surge, tide and current forecasts from the Global Storm Surge Information System (GLOSSIS) at Deltares. This includes 10 day forecasts at hundreds of nearshore locations across the world. See https://www.deltares.nl/en/projects/global-storm-surge-information-system-glossis for more information.",
+      "flowmapLayer":  {
+        "url": "https://storage.googleapis.com/dgds-flowmap/tiles/glossis-currents-latest/{z}/{x}/{y}.png",
```
Comment (evetion) [inline dgds_backend/config_data/datasets.json]: We're only enabling flowmap for the latest timestamp?

Comment (SiggyF) [reply dgds_backend/config_data/datasets.json:29]: no, but I do not know yet how to add  more timestamps

File: datasets/glossis/glossis.py [hunk 8]
```diff
@@ -117,3 +124,20 @@
 
     # Wait for all the tasks to finish
     wait_gee_tasks(taskids)
+
+
+    # This should result in flowmap tiff files
+    flowmap_task_ids = []
+    for current_asset in current_assets:
+        flowmap_tiff = pathlib.Path(current_assets).with_suffix('.tif').name
```
Comment (SiggyF) [inline datasets/glossis/glossis.py:132]: current_asset

File: notebooks/glossis-currents.ipynb (hunk 5 above)
Comment (SiggyF) [reply notebooks/glossis-currents.ipynb:1]: This is  a notebook to try things out.

File: datasets/glossis/glossis.py [hunk 9]
```diff
@@ -117,3 +128,21 @@
...
+
+
+    # This should result in flowmap tiff files
+    flowmap_task_ids = []
+    flowmap_tiffs = []
+    for current_asset in current_assets:
+        flowmap_tiff = pathlib.Path(current_asset).with_suffix('.tif').name
+        flowmap_tiffs.append(flowmap_tiff)
+        task_id = export_flowmap(current_asset, bucket)
+        flowamp_task_ids.append(task_id)
```
Comment (evetion) [inline datasets/glossis/glossis.py:140]: Typo in flow**amp**_task_ids

File: datasets/glossis/glossis.py [hunk 10]
```diff
@@ -76,23 +84,26 @@
         nodata=-9999,
     )
 
+    current_assets = []
```
Comment (evetion) [inline datasets/glossis/glossis.py:87]: Maybe small comment that you need these later to calculate flowmaps?

File: datasets/glossis/glossis.py [hunk 11]
```diff
@@ -117,3 +128,21 @@
...
     # Wait for all the tasks to finish
     wait_gee_tasks(taskids)
+
+
+    # This should result in flowmap tiff files
+    flowmap_task_ids = []
+    flowmap_tiffs = []
+    for current_asset in current_assets:
+        flowmap_tiff = pathlib.Path(current_asset).with_suffix('.tif').name
+        flowmap_tiffs.append(flowmap_tiff)
```
Comment (evetion) [inline datasets/glossis/glossis.py:138]: This is a bit implicit, this will be the names generated from the `export_flowmap` function? Good to introduce usage of pathlib. :)

File: datasets/glossis/glossis.py [hunk 12]
```diff
@@ -117,3 +128,21 @@
...
+        task_id = export_flowmap(current_asset, bucket)
+        flowamp_task_ids.append(task_id)
+    wait_gee_tasks(flowmap_task_ids)
+
+    # This should result in flowmap tiles in a bucket
+    flowmap_tiles_task_ids = []
```
Comment (evetion) [inline datasets/glossis/glossis.py:144]: Were not using this variable, so could be removed.

File: datasets/glossis/glossis.py [hunk 13]
```diff
@@ -117,3 +128,21 @@
...
+    for flowmap_tiff in flowmap_tiffs:
+        tile_dir = generate_wgs84_tiles(flowmap_tiff)
+        upload_dir_to_bucket(bucket_name, source_dir_name=tile_dir, destination_dir_name='flowmaps')
+        # TODO: how do we know which tiles are availabel in backend
```
Comment (evetion) [inline datasets/glossis/glossis.py:148]: Maybe move the TODO (small typo in there) to an issue and remove here?

File: datasets/glossis/utils.py [hunk 14]
```diff
@@ -88,14 +109,14 @@ def upload_to_gee(filename, bucket, asset, wait=True, force=False):
         )
     )
 
-    logging.info(gee_cmd)
-    result = subprocess.run(gee_cmd, shell=True, capture_output=True, text=True)
+    logger.info(gee_cmd)
+    result = subprocess.run(gee_cmd, shell=True, stdout=PIPE, universal_newlines=True)
```
Comment (evetion) [inline datasets/glossis/utils.py:113]: Why the change to `universal_newlines`? You're not using 3.7?

File: datasets/glossis/glossis.py (hunk 13 above)
Comment (SiggyF) [reply datasets/glossis/glossis.py:148]: This is a  separate issue  GDV-324

File: datasets/glossis/utils.py (hunk 14 above)
Comment (SiggyF) [reply datasets/glossis/utils.py:113]: I  was changing python3.6

Comment (SiggyF) [reply datasets/glossis/utils.py:113]: text is an alias for universal_newlines

File: dgds_backend/config_data/datasets.json (hunk 7 above)
Comment (SiggyF) [reply dgds_backend/config_data/datasets.json]: Adding timesteps  is now a separate issue

File: datasets/glossis/glossis.py (hunk 12 above)
Comment (SiggyF) [reply datasets/glossis/glossis.py:145]: comments added
